private/model/api/example_test.go

```diff
@@ -0,0 +1,184 @@
+// +build 1.6,codegen
+
+package api
+
+import (
+	"encoding/json"
+	"testing"
+)
+
+func buildAPI() *API {
+	a := &API{}
+
+	stringShape := &Shape{
+		API:       a,
+		ShapeName: "string",
+	}
+	stringShapeRef := &ShapeRef{
+		API:       a,
+		ShapeName: "string",
+		Shape:     stringShape,
+	}
+
+	intShape := &Shape{
+		API:       a,
+		ShapeName: "int",
+	}
+	intShapeRef := &ShapeRef{
+		API:       a,
+		ShapeName: "int",
+		Shape:     intShape,
+	}
+
+	input := &Shape{
+		API:       a,
+		ShapeName: "FooInput",
+		MemberRefs: map[string]*ShapeRef{
+			"BarShape": stringShapeRef,
+		},
+	}
+	output := &Shape{
+		API:       a,
+		ShapeName: "FooOutput",
+		MemberRefs: map[string]*ShapeRef{
+			"BazShape": intShapeRef,
+		},
+	}
+
+	inputRef := ShapeRef{
+		API:       a,
+		ShapeName: "FooInput",
+		Shape:     input,
+	}
+	outputRef := ShapeRef{
+		API:       a,
+		ShapeName: "Foooutput",
+		Shape:     output,
+	}
+
+	operations := map[string]*Operation{
+		"Foo": &Operation{
+			API:          a,
+			Name:         "Foo",
+			ExportedName: "Foo",
+			InputRef:     inputRef,
+			OutputRef:    outputRef,
+		},
+	}
+
+	a.Operations = operations
+	a.Shapes = map[string]*Shape{
+		"FooInput":  input,
+		"FooOutput": output,
+	}
+	a.Metadata = Metadata{
+		ServiceAbbreviation: "FooService",
+	}
+
+	a.Setup()
+	return a
+}
+
+func TestExampleGeneration(t *testing.T) {
+	example := `
+{
+  "version": "1.0",
+  "examples": {
+    "Foo": [
+      {
+        "input": {
+          "BarShape": "Hello world"
+        },
+        "output": {
+          "BazShape": 1
+        },
+        "comments": {
+          "input": {
+          },
+          "output": {
+          }
+        },
+        "description": "Foo bar baz qux",
+        "title": "I pity the foo"
+      }
+    ]
+  }
+}
+	`
+	a := buildAPI()
+	def := &ExamplesDefinition{}
+	err := json.Unmarshal([]byte(example), def)
+	if err != nil {
+		t.Error(err)
+	}
+	def.API = a
+
+	def.setup()
+	expected := `import (
+"fmt"
+"github.com/aws/aws-sdk-go/aws"
+"github.com/aws/aws-sdk-go/aws/request"
+"github.com/aws/aws-sdk-go/service/fooservice"
+)
+// I pity the foo
+//
+// Foo bar baz qux
+func ExampleFooService_Foo() {
+	svc := fooservice.New(session.New())
+
+	input := FooInput{
+		BarShape: "Hello world",
+	}
+
+	result, err := svc.Foo(input)
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			// Print the error, cast err to awserr.Error to get the Code and
+			// Message from an error.
+			fmt.Println(err.Error())
+		}
+		return
+	}
+
+	fmt.Println(result)
+}
+`
+	if expected != a.ExamplesGoCode() {
+		t.Errorf("Expected:\n%s\n\nReceived:\n%s\n", expected, a.ExamplesGoCode())
```

Comment:
I my PR #1308, I created the utility [`SprintExpectActual`](https://github.com/aws/aws-sdk-go/pull/1308/files#diff-4a2bfc01ae4758c46926349010ae6adeR194) in `awstesting` package. that takes care of this. This should make the testing out put easier to read. Should be able to apply it to all of these large blocks.